### PR TITLE
goals_v4.js has been cleaned up

### PR DESCRIPTION
**Pull request by rybak**
I recommend reviewing each commit individually. Feel free to object to any of the commits – I'll do a rebase and remove them from the pull request. Least objectionable commits (typo fixes and comments) are at the start of the PR.

As discussed in Discord, I left out any tweaks of the `infrequency` field.

### CODE REVIEW

**Comment by rybak** [comment]
Disclaimer: I focused more on Medium, Easy, and Very Easy goals. Changes in Hard and Very Hard are mostly incidental.

**Review by WaterArcanum — APPROVED** [review]

**Comment by rybak** [comment]
Cherry-picked the Rabbit Stew changes from #109 to help avoid dealing with merge conflicts.

**Review by Joshimuz — APPROVED** [review]

**Comment by Joshimuz** [comment]
OK apparently I approve EVERYTHING if I approve one commit cool

**Review by Joshimuz — COMMENTED** [review]
lol

**Comment by Joshimuz** [comment]
Yeah I don't understand how the reviewing stuff works at all

File: goals_v4.js [hunk 1]
```diff
@@ -5,6 +5,7 @@ var Item = {name: "Item", max: [25, 25, 20, 20, 20], line: true},
 Stat = {name: "Stat", max: [5, 5, 5, 5, 5], line: true},
 Action = {name: "Action", max: [20, 20, 20, 20, 20], line: true},
 Build = {name: "Build", max: [20, 20, 20, 20, 20], line: true},
+// note, that on lower difficulties, RareBiome means "Specific Biome"
```
Comment (Joshimuz) [inline goals_v4.js:8]: In v5 maybe we should just make another tag for "Specific Biome"

File: goals_v4.js [hunk 2]
```diff
@@ -57,7 +57,7 @@ var bingoList_v4 = [
 	{name: "(2-5) Bread", antisynergy: ["Bread"], infrequency: 2, tags: [Item, Overworld]},
 	{name: "Hay Bale", antisynergy: ["HayBale"], infrequency: 2, tags: [Item, Overworld]},
 	{name: "Flower Pot", tags: [Item, Overworld]},
-	{name: "(2-10) Feathers", antisynergy: ["Feathers"], infrequency: 2, tags: [Item]},
+	{name: "(2-10) Feathers", antisynergy: ["Feathers"], infrequency: 2, tags: [Item, Overworld]},
```
Comment (Joshimuz) [inline goals_v4.js:60]: Yes you can, Zomibfied Piglins can spawn on Chickens as Chicken Jockeys

File: goals_v4.js [hunk 3]
```diff
@@ -153,7 +153,7 @@ var bingoList_v4 = [
 	{name: "Fermented Spider Eye", tags: [Item, Overworld]},
 	{name: "(5-7) Different Stairs", antisynergy: ["Stairs"], infrequency: 2, tags: [Item]},
 	{name: "(2-3) Ender Pearls", antisynergy: ["EnderPearl"], infrequency: 2, tags: [Item]},
-	{name: "16 Chicken Eggs", tags: [Item]},
+	{name: "16 Chicken Eggs", tags: [Item, Overworld]},
```
Comment (Joshimuz) [inline goals_v4.js:156]: Zomibfied Piglins can spawn on Chickens as Chicken Jockeys

**Comment by Joshimuz** [comment]
Other than the Chickens in Nether thing everything else looks good!

**Comment by rybak** [comment]
Force pushed after a rebase to drop the commits, which added tag Overworld to the Chicken-related goals. 

Range-diff:
```
$ git range-diff efd6276 e58b41a 668a7e2
 1:  147ef40 =  1:  147ef40 goals_v4: fix typo in UseFurnace catalyst
 2:  edadff7 =  2:  edadff7 goals_v4: fix typos in the "Ink Sacs" goal
 3:  77aa1aa =  3:  77aa1aa goals_v4: fix typo in `antisynergy`
 4:  e6d5046 =  4:  e6d5046 goals_v4: sort fields of "(32-64) Stone"
 5:  8d52752 =  5:  8d52752 goals_v4: clarify usage of tag RareBiome on lower difficulties
 6:  8cceddc =  6:  8cceddc goals_v4: explain 3 Different Edible Items not having an Overworld tag
 7:  efa2da1 =  7:  efa2da1 goals_v4: explain Mossy Cobblestone not having tag RareBiome
 8:  b299787 =  8:  b299787 goals_v4: remove reactant "Death"
 9:  46db08d =  9:  46db08d goals_v4: remove tag Action from non-action Never goals
10:  f73f885 = 10:  f73f885 goals_v4: remove tag Overworld from Golden Apple goals
11:  0d601da = 11:  0d601da goals_v4: add tag Overworld to Bookshelves goal
12:  66f27eb <  -:  ------- goals_v4: add tag Overworld to Feathers goals
13:  a1e12de <  -:  ------- goals_v4: add tag Overworld to Chicken Eggs goal
14:  ce917bd = 12:  94f6409 goals_v4: add tag Overworld to goals requiring Sand
15:  c9e6a8f = 13:  901883e goals_v4: clean up name of the Melon Slices goal
16:  a56b593 = 14:  961af79 goals_v4: add antisynergies for all Bucket types
17:  4601045 = 15:  8588437 goals_v4: add antisynergy for "Enchant N item(s)"
18:  d658b66 = 16:  4634c57 goals_v4: add reactant Pacifist and tag Combat to Potions
19:  941d09a = 17:  182c330 goals_v4: add reactant Pacifist to "Become Hero of the Village"
20:  0fc5a8d = 18:  56bdf8f Remove Pacifist and RareBiome from Rabbit Stew
21:  e58b41a = 19:  668a7e2 Remove Nether tag in "Ice Block on a Magma Block"
```

**Pull request merged**